### DISCUSSION TITLE
Why does this result page does not count failed expectation test? But if I send true values it shows it in the resulting page.

### DISCUSSION THREAD

**Discussion by almatkai** [great-expectations/great_expectations#8908] [Support]
```
import great_expectations as gx

context = gx.get_context()

validator = context.sources.pandas_default.read_csv(
    "data.csv"
)

res1 = validator.expect_column_values_to_not_be_null("pickup_datetime")
# This one below with Error(true => min=1, max=6)
res2 = validator.expect_column_values_to_be_between("passenger_count", min_value=1, max_value=5)
validator.save_expectation_suite()

checkpoint = context.add_or_update_checkpoint(
    name="my_quickstart_checkpoint",
    validator=validator,
)

checkpoint_result = checkpoint.run()
context.view_validation_result(checkpoint_result)
```
![CleanShot 2023-11-01 at 12 49 55@2x](https://github.com/great-expectations/great_expectations/assets/92521753/ee1c61ed-2057-4db4-ad27-2c573e2333db)

**Comment by michalc** [comment]
(I have literally 1 day of experience with Great Expectations)

This issue just surprised us - we were trying to make a failing expectation to make sure the system worked, and it would just disappear.

To fix this, we changed:

```python
validator.save_expectation_suite()
```
to include `discard_failed_expectations=False`
```python
validator.save_expectation_suite(discard_failed_expectations=False)
```
It's in the docs at: https://docs.greatexpectations.io/docs/reference/api/validator/validator/Validator_class#great_expectations.validator.validator.Validator.save_expectation_suite

I'm starting to suspect something: the documentation assumes you have a "correct" dataset that you use to interactively develop expectations with. It's not really our use case though - we might have "wrong data", or even in fact, we might not even have _any_ data at all.

(Right now we're kinda wrestling with trying to work out how to use Great Expectations for our case, if at all - the assumptions/workflows don't quite seem aligned with our world... Do we adjust our workflows, or can we figure out how to use Great Expectations within them?)
